packages/authorization-process/test/getClients.test.ts

```diff
@@ -0,0 +1,267 @@
+import {
+  Client,
+  PurposeId,
+  TenantId,
+  generateId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { describe, expect, it } from "vitest";
+import { getMockClient, getRandomAuthData } from "pagopa-interop-commons-test";
+import { genericLogger } from "pagopa-interop-commons";
+import { addOneClient, authorizationService } from "./utils.js";
+
+describe("getClients", async () => {
+  const consumerId = generateId();
+  const purposeId = generateId();
+  const mockClient1: Client = {
+    ...getMockClient(),
+    name: "test1",
+    consumerId: unsafeBrandId(consumerId),
+    kind: "Consumer",
+  };
+  const mockClient2: Client = {
+    ...getMockClient(),
+    name: "test2",
+    consumerId: unsafeBrandId(consumerId),
+    kind: "Consumer",
+  };
+
+  const userId1 = generateId();
+  const userId2 = generateId();
+  const mockClient3: Client = {
+    ...getMockClient(),
+    users: [unsafeBrandId(userId1), unsafeBrandId(userId2)],
+    consumerId: unsafeBrandId(consumerId),
+  };
+  const userIds3 = generateId();
+  const userIds4 = generateId();
+  const mockClient4: Client = {
+    ...getMockClient(),
+    users: [unsafeBrandId(userIds3), unsafeBrandId(userIds4)],
+    consumerId: unsafeBrandId(consumerId),
+  };
+
+  const mockClient5: Client = {
+    ...getMockClient(),
+    purposes: [unsafeBrandId<PurposeId>(purposeId)],
+    consumerId: unsafeBrandId(consumerId),
+  };
+
+  const mockClient6: Client = {
+    ...getMockClient(),
+    purposes: [unsafeBrandId<PurposeId>(purposeId)],
+    consumerId: unsafeBrandId(consumerId),
+  };
+
```

Comment:
one test for [this](https://github.com/pagopa/interop-be-monorepo/pull/563/files#diff-dbdfef4027ce7e4ab4a5697de6c66e5364588bf66e77ff0558e90d05e496dfd2R132-R134) logic would be very helpful